common/guestchain/src/manager.rs

```diff
@@ -168,6 +181,43 @@ impl<PK: crate::PubKey> ChainManager<PK> {
         self.pending_block.as_ref()
     }
 
+    pub fn update_config(
```

Comment:
I think I’d actually make it a method of Config, i.e. Config::update.
Then UpdateConfigError could also be defined in config.rs.  And then
this method would just be

```rust
 pub fn update_config(
        &mut self,
        update: UpdateChainConfigPayload,
    ) -> Result<(), UpdateConfigError> {
        self.config.update(update)
    }
```

Not sure if that’s better, but it would move code out of manager.rs
into config.rs resulting in better balance. ;)

Reply:
ya i think that should be possible.